chaos/cluster/cluster.go

```diff
@@ -0,0 +1,219 @@
+package cluster
+
+import (
+	"bytes"
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	"golang.org/x/crypto/ssh"
+
+	"github.com/coreos/ktestutil/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+const (
+	cmdReboot          = "sudo ifconfig eth0 down && sudo systemctl stop kubelet && sudo reboot"
+	cmdRebootWithSleep = "sudo ifconfig eth0 down && sudo systemctl stop kubelet && sleep %ds && sudo reboot"
+	cmdSystemUp        = "sudo systemctl is-active kubelet"
+)
+
+// Cluster is a simple abstraction that stores cluster nodes.
+// It allows rebooting the entire cluster / nodes.
+type Cluster struct {
+	Masters []*v1.Node
+	Workers []*v1.Node
+	// Defaults to 0.
+	RebootAfter time.Duration
```

Comment:
I still feel this para should not belong to cluster struct. we should specify this when calling reboot.